### PR TITLE
Segment API: category info and search by category alias

### PR DESCRIPTION
**Pull request by patrykgruszka**
| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | "features"
| Bug fix?                               | no
| New feature?                           | no
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | yes
| Related user documentation PR URL      | https://github.com/mautic/mautic-documentation/pull/209
| Related developer documentation PR URL | https://github.com/mautic/developer-documentation/pull/199
| Issue(s) addressed                     | Fixes #... 

#### Description:
This pull requests provides category info in segment API data and adds standard category search by query `category:category-alias`.

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Create segment with category and make API request to view data. Category info should be available `GET /api/segments/{YOUR_SEGMENT_ID}`
3. Test search by category alias though API `/api/segments?search=category:{YOUR_CATEGORY_ALIAS}`. Search can be also tested in segment list in Mautic panel by query `category:category-alias`.

### CODE REVIEW

File: app/bundles/LeadBundle/Tests/Controller/Api/ListApiControllerFunctionalTest.php
Comment (escopecz) [inline app/bundles/LeadBundle/Tests/Controller/Api/ListApiControllerFunctionalTest.php]: We know there is one segment with the kitty category. I think we can test for that rather than not empty.
```suggestion
        $this->assertCount(1, $response['lists']);
```

File: app/bundles/LeadBundle/Tests/Controller/Api/ListApiControllerFunctionalTest.php
Comment (escopecz) [inline app/bundles/LeadBundle/Tests/Controller/Api/ListApiControllerFunctionalTest.php]: There is a shorter way how to write the same:
```suggestion
        $this->assertTrue($clientResponse->isOk());
```

File: app/bundles/LeadBundle/Tests/Controller/Api/ListApiControllerFunctionalTest.php
Comment (escopecz) [inline app/bundles/LeadBundle/Tests/Controller/Api/ListApiControllerFunctionalTest.php]: There is a shorter way how to write the same:
```suggestion
        $this->assertTrue($clientResponse->isOk());
```

File: app/bundles/LeadBundle/Tests/Controller/Api/ListApiControllerFunctionalTest.php
Comment (escopecz) [inline app/bundles/LeadBundle/Tests/Controller/Api/ListApiControllerFunctionalTest.php]:
```suggestion
    public function testSegmentWithCategory(): void
```

**Review by escopecz — DISMISSED** [review]
This is great. Thanks for the tests! I have minor suggestions. Please look at them but I approve this PR even if you won't accept them. This PR is ready either way.

**Comment by escopecz** [comment]
Could you also please make a quick PR with your changes into https://github.com/mautic/developer-documentation/blob/master/source/includes/_api_endpoint_segments.md?

**Comment by patrykgruszka** [comment]
@escopecz thanks for suggestions and review. I've submitted update to the developers documentation https://github.com/mautic/developer-documentation/pull/199.

**Review by RCheesley — DISMISSED** [review]
Thanks for the PR and the clear testing instructions @patrykgruszka !

Confirmed that the category info is now included:

screenshot-nimbus-capture-2021 05 23-11_12_34

Also that it is possible to query for the segments based on category:

screenshot-nimbus-capture-2021 05 23-11_17_20

and also in the search panel on the Segments page:

screenshot-mautic4 ddev site-2021 05 23-11_18_12

Small suggestion - we should add it to this list of available search terms in the front-end UI. This could be done in a follow up PR and should not block merging this PR in time for the release (will break it out into a separate issue)

screenshot-mautic4 ddev site-2021 05 23-11_18_55

I will make a PR to the end-user docs to add it here: https://docs.mautic.org/en/mautic-search#segments

**Comment by RCheesley** [comment]
Docs PR: https://github.com/mautic/mautic-documentation/pull/209/

**Comment by RCheesley** [comment]
Just pushed the commit to add the alias to the available search operators so that we can get this merged for the release 🚀

**Comment by RCheesley** [comment]
@all-contributors please add @patrykgruszka for code

**Comment by RCheesley** [comment]
@all-contributors please add @escopecz for Review

**Comment by RCheesley** [comment]
@all-contributors please add @patrykgruszka for doc

**Comment by RCheesley** [comment]
@all-contributors please add @rcheesley for doc

**Pull request merged**
